oci/casext/gc.go

```diff
@@ -25,6 +25,9 @@ import (
 	"golang.org/x/net/context"
 )
 
+// GCPolicy is a policy function that returns if a blob can be GC'ed
+type GCPolicy func(desc ispec.Descriptor) (bool, error)
```

Comment:
In line with switching to `Paths`, this should be a `DescriptorPath`.

Reply:
Want to hide umoci internals from the policy func writers. So just pass in the standard ispec.Descriptor.

Reply:
`DescriptorPath` isn't really an umoci internal, it's fairly publicly used if you want to do `ResolveReference` (so basically every umoci API user will see `DescriptorPath` at some point). For a bit of background -- the reason why `DescriptorPath` exists is that it allows programs to correctly handle arbitrary nesting of descriptors.

*However* I just re-read the original description of the PR and realised that passing descriptors makes no sense -- you are only calling `policy(desc)` if there is no descriptor in the black set (in other words, on blobs that have no descriptor pointing to them), so the descriptor will always be an empty struct. The new tests didn't catch this, so you'll need to update the tests so they can exercise this code properly.